zkevm-circuits/src/evm_circuit/execution/exp.rs

```diff
@@ -184,7 +184,7 @@ impl<F: Field> ExecutionGadget<F> for ExponentiationGadget<F> {
         );
 
         // Finally we build an expression for the dynamic gas cost.
-        let dynamic_gas_cost = 50.expr() * exponent_byte_size.byte_size();
+        let dynamic_gas_cost = 50.expr() * exponent_byte_size.byte_size_expr();
```

Comment:
replace 50 with `OpcodeId::EXP.constant_gas_cost().0` ?

Reply:
LGTM with some comments.

Reply:
`OpcodeId::EXP.constant_gas_cost().0 = 10`, it is `static_gas`.

Added a new constant `GasCost::EXP_BYTE_TIMES = 50` (and replaced in both ExpGadget and OogExpGadget).
And add the below comments for gas cost calculation.
```
gas_cost = static_gas + dynamic_gas
static_gas = 10
dynamic_gas = 50 * exponent_byte_size
```